\section{Discussion}

\subsection{In what sense is this ultimate?}
Calling this set of rotamer-library distributions "ultimate" is a claim that requires both explanation and justification, beyond the mere fact that it follows after our "penultimate" rotamer library. 

First, we claim it as ultimate for the rotamer aspect of model validation (where outlier status is pre-eminent)  -- such as done by the MolProbity website, in \textit{PHENIX}, or by the wwPDB -- but not necessarily for other purposes (such as protein design) that depend on accurate comparisons within the regions of high probability rather than on accurate delineation of the outer limits of those regions.

Second, changes from the penultimate distributions are fairly minor in quantitative terms;  their importance lies in the new ability to support a 3-level evaluation, to identify very rare rotamers, and to specify rotamer-dependent bond-angle deviations. With a million residues of high-reliability data, the sampling is now adequate for this task. The results are unlikely to change much in the future, except for some special cases such as disulfides or chemically modified amino acids. The multi-dimensional distributions of rotamer probability are freely available on GitHub for use in other software systems.

Third, what we do expect will improve in the future is a change in how sidechain modeling and validation is defined, driven both by expansion of data and compute power and by the needs of structure determination at lower resolution. Rather than separate Ramachandran and rotamer evaluation, we should move toward analyzing all backbone and sidechain torsional dimensions together \cite{JSR_theplot_2013}, including allowance for the influence of secondary structure and local motifs. Thus our "ultimate" claim asserts the position that rotamer evaluation should no longer be updated but should evolve into something else.

\subsection{Use of the Top8000 rotamer distributions in model building}
Although the Top8000 rotamer distributions were created specifically for MolProbity's validation of sidechain conformations, they also can be used in model building for crystallography, design, homology, etc., as the Top500 library has been \cite{Langer2008, Emsley:ba5144, Headd2008, Terwilliger2008, Adams:2010fk, Gainza2013}. To make model building computationally tractable, sidechains are initially fit as discrete conformations at rotamer values, then either minimized or tested at neighboring sample points using $\sigma$ to inform the allowable range for each $\chi$. However, most rotamer distributions are far from normal and often have quite complex shapes, as can be seen in Figures \ref{fig:AspAsn} and \ref{fig:outcontour}. Thus the approach of using a central value and $\sigma$ cannot adequately describe allowed sidechain conformations. 

The contour values in our rotamer distributions correspond to the percentage of high-quality reference data that lies outside that contour. The consequence of this treatment is that the contour data represent a probability density function in $\chi$ space describing sidechain conformations. We recommend using those contours to delimit sampling or minimization in $\chi$ space, with the added benefit that the contour levels provide prior probabilities for each position. If a contour level higher than 0.3\% is chosen for this purpose, then rotamers which do not reach the chosen percentage level should be omitted. Depending on details of the algorithm used, it may or may not be beneficial to model significant rotamer-dependent bond-angle deviations.

\subsection{I have a rotamer outlier; is it wrong?}
An accurate macromolecular model gives unparalleled mechanistic knowledge at the single-molecule level. Many such models, of different cellular components, combined with structural perturbations such as point mutations and ligand binding, provide mechanistic detail on the cellular level. These methods not only provide interesting knowledge, they allow understanding crucial to the treatment of disease. The utility of a model in this process is correlated with how accurate it is. As such, structural biologists want to build models that come as close as possible to representing a valid state of the actual macromolecule. Unfortunately, the experimental data alone seldom provide all information needed to build an adequate model. Fortunately, empirical knowledge of chemical and macromolecular structure helps greatly, and in crystallography much of this knowledge (e.g. bond lengths and angles, chirality) is already part of the automated software. Structure validation such as MolProbity provides even more empirical knowledge, often highlighting errors that refinement was unable to fix.

Sometimes we get disturbing reports that people are trying to achieve a sort of "MolProbity Nirvana" by attempting to eliminate every single outlier. To approach eliminating all clashes is a worthy goal: in principle the clash target is zero, as sterics do not allow serious atomic overlaps. However, at high resolution a zero score would require the difficult reconciliation of occupancies for alternate conformations including waters, and in any case a few puzzles often remain. Another important point to keep in mind is that the primary purpose and usefulness of model validation is to help diagnose and correct places where the conformation has been fit in the wrong local energy well. Such corrections often matter to biological interpretation, and will be stable to further refinement. In contrast, a small shift across the border into the allowed region will often shift right back again, and is not a very meaningful improvement.

However, the target is very definitely \textit{not} zero for torsion validation. The idea that \textit{every single} Ramachandran and rotamer conformation needs to be allowed is a misunderstanding. As outlined here, validation of torsion angles relies on contoured conformational distributions from quality-filtered datasets. The outer contour defines an outlier cutoff, meaning that even the filtered reference data has outliers but \textit{not} that those conformations are wrong. There is room for valid torsion outliers, more so for sidechain rotamers than for backbone Ramachandran values. Even in 2005 the curve of rotamer-outlier percentage, fit as a function of resolution, was found to asymptote at 0.5\% \cite{Arendall2005}. For quality-filtered data the rotamer outlier cutoff is now 0.3\%, meaning that 3 out of 1,000 residues are expected to be valid rotamer outliers. For Ramachandran the outlier cutoff is 0.05\%, so only a very low 5 out of 10,000 residues -- but not zero -- are expected to be valid Ramachandran outliers. In either case, to be accepted as valid, an outlier should have clear electron density to support its occurrence, and should have either H-bonds or tight packing to hold it in the presumably quite strained conformation. This is exemplified by the Asn rotamer outliers in Figure \ref{fig:AsnOutliers}, which each have an eclipsed $\chi$1 but are validated by excellent density and three sidechain H-bonds.  Valid outliers are more likely to have been selected and maintained by evolution if there is a functional need of some sort (folding, catalysis, binding) for that specific conformation; therefore such cases will probably be of great interest and will reward as well as require detailed examination.
